### PR TITLE
first attempt

### PR DESCRIPTION
**Pull request by maximz101**
#### Check List:
- [x] Tests pass (`./test.sh <username>` shows no differences between expected and actual outputs)
- [x] All formatting changes by the build are committed
- [x] Your launch script is named `calculate_average_<username>.sh` (make sure to match casing of your GH user name) and is executable
- [x] Output matches that of `calculate_average_baseline.sh`

* Specs : MBP M2, 12 Cores, 32GB RAM
* Execution time: 0m27.7s
* Execution time of reference implementation: 3m3s

### CODE REVIEW

**Comment by gunnarmorling** [comment]
There's a difference off the output to the baseline script. Please run ./calculate_average_baseline.sh and make sure your output matches it.

File: calculate_average_maximz101.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+# Uncomment below to use sdk
+source "$HOME/.sdkman/bin/sdkman-init.sh"
+sdk use java 21.0.1-amzn 1>&2
+
+JAVA_OPTS="--enable-preview -Xmx6g"
+time java $JAVA_OPTS --class-path target/average-1.0.0-SNAPSHOT.jar dev.morling.onebrc.CalculateAverage_maximz101
```
Comment (gunnarmorling) [inline calculate_average_maximz101.sh:23]:
```suggestion
java $JAVA_OPTS --class-path target/average-1.0.0-SNAPSHOT.jar dev.morling.onebrc.CalculateAverage_maximz101
```

Not needed anymore.

File: calculate_average_maximz101.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+# Uncomment below to use sdk
+source "$HOME/.sdkman/bin/sdkman-init.sh"
+sdk use java 21.0.1-amzn 1>&2
```
Comment (gunnarmorling) [inline calculate_average_maximz101.sh:20]: Could you move this please into a new script prepare_maximz101.sh. We have changed the organization of the files, so as to keep this call out of the evaluated run itself. Thx!

**Comment by maximz101** [comment]
I've updated the code to adapt to the new structure. I have double checked the output and run the test script.
Here's my run result on a the same mbp M2/12 core/32GB

```
❯ hyperfine ./calculate_average_maximz101.sh
Benchmark 1: ./calculate_average_maximz101.sh
  Time (mean ± σ):      7.920 s ±  0.209 s    [User: 82.796 s, System: 1.713 s]
  Range (min … max):    7.666 s …  8.387 s    10 runs

```

Thanks!!

**Comment by gunnarmorling** [comment]
00:18.902, thx for participating!

**Pull request merged**